src/services/cart.service.js

```diff
@@ -0,0 +1,56 @@
+import Products from '../database/models/products.model';
+// eslint-disable-next-line import/no-cycle
+import { redisClient } from '../helpers';
+
+async function createCart(userId) {
+  const products = await redisClient.set(`cart_${userId}`, JSON.stringify([]));
+  if (products) {
+    return JSON.parse(products);
+  }
+}
+
+async function getProductsInCart(userId) {
+  if (!(await redisClient.get(`cart_${userId}`))) await createCart(userId);
+  const redisProducts = await redisClient.get(`cart_${userId}`);
+  const result = JSON.parse(redisProducts);
+  return result;
+}
+
+async function getProductPrice(productId) {
+  const product = await Products.findOne({ where: { id: productId } });
+  return product.price;
+}
+
+async function displayCart(userId) {
+  const newCart = await getProductsInCart(userId).then((data) => data);
+  async function totalPrice(arr) {
+    let sum = 0;
+    for (let i = 0; i < arr.length; i += 1) {
+      // eslint-disable-next-line no-await-in-loop
```

Comment:
you may use a promise in stead of using `await `in a `loop `. 
at certain time it may misbehave when done like this